src/dog.sol

```diff
@@ -148,14 +149,16 @@ contract Dog /* is LibNote */ {
             require(room > 0 && room >= dust, "Dog/liquidation-limit-hit");
 
             dart = min(art, mul(room, WAD) / rate / milk.chop);
+
+            // Verify that CDP is not left in a dusty state
+            require(dart == art || mul(art - dart, RAY) >= dust, "Dog/leaves-dust");
```

Comment:
Need to multiply by `rate` here, not `RAY`, because `dust` is measured in dai, not in normalized debt.